src/store/forms.ts

```diff
@@ -29,7 +29,24 @@ export const mutateForm = (
     form: NodeEditorForm,
     toMerge: Partial<NodeEditorForm>,
     toRemove: string[] = []
-): NodeEditorForm => mutate(form || {}, { $merge: toMerge, $unset: toRemove }) as NodeEditorForm;
+): NodeEditorForm => {
+    let updated = mutate(form || {}, { $merge: toMerge, $unset: toRemove }) as NodeEditorForm;
+    let valid = true;
+    for (const key of Object.keys(updated)) {
+        const entry: any = updated[key];
+        if (typeof entry === 'object') {
+            if (entry.validationFailures && entry.validationFailures.length > 0) {
+                valid = false;
+                break;
+            }
+        }
+    }
```

Comment:
`updateForm` is a bit smarter in that it inspects keys for validation failures and sets the overall form validity from that